src/Pages/VideoPlayer/VideoPlayer.jsx

```diff
@@ -0,0 +1,170 @@
+import { useParams } from "react-router-dom";
+import { useEffect } from "react";
+import { Navbar, Sidebar, Loader } from "../../Components";
+import { useData, useAuth } from "../../Contexts";
+import ThumbUpIcon from "@material-ui/icons/ThumbUp";
+import WatchLaterIcon from "@material-ui/icons/WatchLater";
+import "./VideoPlayer.css";
+import { updatePlaylist } from "../../services/dataServices";
+
+export function VideoPlayer() {
+  const { videoId } = useParams();
+  const { currentUser } = useAuth();
+
+  const { data, dispatch, playlists } = useData();
+  const video = data.find((item) => item.videoId === videoId);
+  const { title, description, uploadedBy, likes, views, subscribers } = video;
+
+  console.log("Playlists:", playlists);
+
+  const isVideoInPlaylist = (playlistId, videoId) => {
+    console.log(
+      playlists
+        .find((list) => list.id === playlistId)
+        .videos.find((video) => video.videoId === videoId)
+    );
+    return playlists
+      .find((list) => list.id === playlistId)
+      .videos.find((video) => video.videoId === videoId);
+  };
+
+  const updatePlaylistInDb = async (playlistId, video) => {
+    console.log("Update this vode in db", video._id, video.videoId);
+    const response = await updatePlaylist(
+      currentUser.userId,
+      playlistId,
+      video._id
+    );
+    console.log(response);
+    return response.data.success;
+  };
+
+  useEffect(() => {
+    currentUser &&
+      (async function () {
+        if (
+          !playlists
+            .find((list) => list.id === "history")
+            .videos.find((video) => video.videoId === videoId)
+        ) {
+          const response = await updatePlaylist(
+            currentUser.userId,
+            "history",
+            video._id
+          );
+          response
+            ? dispatch({
+                type: "ADD_TO_PLAYLIST",
+                payload: { playlistId: "history", video: video },
+              })
+            : console.error("Error while updating in DB");
+        }
+      })();
+  }, [currentUser, dispatch, playlists, video, videoId]);
```

Comment:
1. Hey rohit, the video titled `The NFT Bubble: Why 97% of All NFT Art Right Now Is a Bad Investment | The Outer Realm Podcast` was two times in my history when i was logged in, you might want to check this out.

2. You can include try catch block here in useEffect and destructure the response which you are getting for readability concerns